deeprank/features/FullPSSM.py

```diff
@@ -212,6 +212,14 @@ def get_feature_value(self, cutoff=5.5):
                 f"{self.mol_name}: The following interface residues have "
                 f" no pssm value:\n {ctc_res_wo_pssm}"
             )
+        elif len(pssm_res_set.difference(ctc_res_set)) > 0:
```

Comment:
The `pssm_res_set` contains pssm for all residues, and `ctc_res_set` consider only interface residues, so this condition should always be True.
Also `pssm_res_wo_ctc = pssm_res_set.difference(ctc_res_set)` misstreats the non-interface residues as residues with missing atoms.

Reply:
The issue is that some of the res are missing in xyz_dict therefore leading to a keyError  on line 229. I think we should create a `get_residue_centers` either in DeepRank or pdb2sql. There we could specify to get either CB, or CA if CB is missing or take the mean of the residue xyz if CA and CB are missing. The only issue is the we would have to loop over the residue and it will be a bit slow for large complexes ....

Reply:
I think we need a data preprocess step in DeepRank to avoid these problems.   We can could add a PDB validator in `pdb2sql`.

Reply:
that's a good idea ... but I wouldn't know how to do it

Reply:
I've introduced a get_residue_center in FetureClass I think we can live with that for now ....

Reply:
Thanks, the `get_residue_center` looks good.